src/image_modifications.cpp

```diff
@@ -141,9 +142,7 @@ modification_queue modification::decode(const std::string& encoded_mods)
 	modification_queue mods;
 
 	for(const std::string& encoded_mod : utils::parenthetical_split(encoded_mods, '~')) {
-		if(auto mod = decode_modification(encoded_mod)) {
-			mods.push(std::move(mod));
-		}
+		mods.push(decode_modification(encoded_mod));
```

Comment:
Why have you changed this to put null pointers in there?

Reply:
Redundant. Push already checks for non-null.

Reply:
Looks like you're technically correct, but it still _feels_ wrong. A function that pushes the value only if it's not null really has no business being called `push`… it's not doing what it says it does…

I think I'd prefer to have the redundant check in this case, with a very visible error message in `push` if it somehow ends up receiving a null pointer (since as the comment says, that shouldn't happen).

Reply:
I'll just move the check here. You're right API-wise that a push function should push.